app/src/androidTest/java/com/github/h3lp3rs/h3lp/forum/ForumAnswersActivityTest.kt

```diff
@@ -1,83 +1,68 @@
 import com.github.h3lp3rs.h3lp.forum.ForumPostsActivity.Companion.selectedPost
-import com.github.h3lp3rs.h3lp.forum.data.ForumPostData
 import com.github.h3lp3rs.h3lp.signin.SignInActivity.Companion.setName
 import junit.framework.Assert.assertEquals
-import org.junit.After
 import org.junit.Before
 import org.junit.Test
 import org.junit.runner.RunWith
-import org.mockito.Mockito.mock
-import org.mockito.kotlin.any
-import java.time.ZonedDateTime
-import org.mockito.Mockito.`when` as When
-import com.github.h3lp3rs.h3lp.forum.ForumCategory.GENERAL
 
-const val CATEGORY_TEST_STRING = "GENERAL"
-val CATEGORY_TEST = GENERAL
+const val CATEGORY_TEST_STRING = "TRAUMATOLOGY"
 const val QUESTION_TEST = "question"
 const val ANSWER_TEST = "answer"
 
 @RunWith(AndroidJUnit4::class)
 class ForumAnswersActivityTest {
-    private val forumPosts : MutableMap<String, List<String>> = mutableMapOf()
+    private lateinit var forum: Forum
+
+    private val launchIntent = Intent(
+        ApplicationProvider.getApplicationContext(), ForumAnswersActivity::class.java
+    ).apply {
+        putExtra(EXTRA_FORUM_CATEGORY, CATEGORY_TEST_STRING)
+    }
 
     @Before
     fun setup() {
-        val intent = Intent(
-            ApplicationProvider.getApplicationContext(), ForumAnswersActivity::class.java
-        ).apply {
-            putExtra(EXTRA_FORUM_CATEGORY, CATEGORY_TEST_STRING)
-        }
-
         setName(USER_TEST_ID)
-
-        val forum = mock(Forum::class.java)
-        setForum(CATEGORY_TEST, forum)
-        When(forum.newPost(any(), any())).then {
-            val content = it.getArgument<String>(1)
-            forumPosts[QUESTION_TEST] = listOf(content)
-            any()
-        }
-        When(forum.child(any() as String)).thenReturn(forum)
-        When(forum.child(any() as Path)).thenReturn(forum)
-
-        selectedPost = ForumPost(
-            forum, ForumPostData(
-                "", QUESTION_TEST, ZonedDateTime.now(), "", "", GENERAL
-            ), emptyList()
-        )
-
-        ActivityScenario.launch<ForumAnswersActivity>(intent)
-
-        init()
-    }
-
-    @After
-    fun clean() {
-        release()
+        mockForum()
+        forum = forumOf(TRAUMATOLOGY)
     }
 
     @Test
     fun addNewAnswerWorks() {
-        onView(withId(R.id.text_view_enter_answer)).perform(replaceText(ANSWER_TEST))
+        forum.newPost("", QUESTION_TEST).thenAccept { post ->
+            selectedPost = post
 
-        onView(withId(R.id.add_answer_button)).perform(click())
+            launch<ForumAnswersActivity>(launchIntent).use {
+                // Add an answer to post
+                onView(withId(R.id.text_view_enter_answer)).perform(replaceText(ANSWER_TEST))
+                onView(withId(R.id.add_answer_button)).perform(click())
+
+                // No other way to do this, since delayed futures are not supported by Kotlin
+                // --> java.lang.NoSuchMethodError when using a delayed executor
```

Comment:
That's some bad news :, (
one solution I see is that since this file contains only one test  we could code this test in java but that will not be really coherent

Reply:
Probably there is a way to update Kotlin guava or something like that to make these classes work. But that's a bit risky to do one sprint before the final deadline :(